kalite/distributed/management/commands/retrievecontentpack.py

```diff
@@ -174,23 +174,25 @@ def extract_assessment_items(zf, lang):
 
     assessment_dest_dir = os.path.join(content_settings.ASSESSMENT_ITEM_ROOT, channel)
 
-    ensure_dir(assessment_dest_dir)
+    # clear old assessment items, so we consume less space during the extraction process
+    _remove_old_assessment_resources(assessment_dest_dir)
 
-    items = (s for s in zf.namelist() if assessment_zip_dir in s)
+    _extract_assessment_resources(zf, assessment_zip_dir, assessment_dest_dir)
 
-    for item in items:
-        # files inside zipfiles may come with leading directories in their
-        # names, like subtitles/hotdog.vtt. We'll only want the actual filename
-        # (hotdog.vtt) when extracting as that's what KA Lite expects.
 
-        filename = os.path.basename(item)
-        subfolder = filename[:3]
-        assessment_subfolder = os.path.join(assessment_dest_dir, subfolder)
+def _remove_old_assessment_resources(assessment_dir):
+    shutil.rmtree(assessment_dir)
+
+
+def _extract_assessment_resources(zf, assessment_zip_dir, assessment_dest_dir):
+    tempdir = tempfile.mkdtemp()
+
+    items = (s for s in zf.namelist() if assessment_zip_dir in s)
+
+    zf.extractall(tempdir, items)
 
-        ensure_dir(assessment_subfolder)
+    temp_assessment_dir = os.path.join(tempdir, assessment_zip_dir)
 
-        zip_item_fileobj = zf.open(item)
+    shutil.move(temp_assessment_dir, assessment_dest_dir)
```

Comment:
@benjaoming that's why we delete the destination directory first!

Although now that I think about it, we'll probably want to see if assessment resources exist in the zip file before doing the deletion. Else we might end up with the user not being able to load any exercises at all.